extension.js

```diff
@@ -1,5 +1,7 @@
 'use strict';
 
+var fs = require('fs');
+var path = require('path');
 var vscode = require('vscode');
 var fileService = require('./libs/fileService');
```

Comment:
You should be able to use the existing fileService or modify it if necessary, rather then rewrite a lot of its functionality. The 'Open File...' context menu should just add the option of the UI passing a file location to the fileService to open, in addition to the existing default behaviour of passing the current active document file location to the fileService to open activated by Command/CTRL+ALT+O.

There is another pull request #5 with similar functionality. You may wish to wait for the outcome of that to avoid duplicating efforts.